packages/core/src/customer/strategies/paypalcommerce/paypalcommerce-customer-strategy.ts

```diff
@@ -0,0 +1,411 @@
+
+    private async _onShippingOptionsChange(
+        data: ShippingOptionChangeCallbackPayload,
+    ): Promise<void> {
+        const shippingOption = this._getShippingOptionOrThrow(data.selectedShippingOption.id);
+
+        try {
+            await this._store.dispatch(
+                this._consignmentActionCreator.selectShippingOption(shippingOption.id),
+            );
+            await this._updateOrder();
+        } catch (error) {
+            this._handleError(error);
+        }
+    }
+
+    private async _submitPayment(methodId: string, orderId: string): Promise<void> {
+        const paymentData = {
+            formattedPayload: {
+                vault_payment_instrument: null,
+                set_as_default_stored_instrument: null,
+                device_info: null,
+                method_id: methodId,
+                paypal_account: {
+                    order_id: orderId,
+                },
+            },
+        };
+
+        await this._store.dispatch(
+            this._paymentActionCreator.submitPayment({ methodId, paymentData }),
+        );
+    }
+
+    private async _updateOrder(): Promise<void> {
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        const consignment = state.consignments.getConsignmentsOrThrow()[0];
+
+        try {
+            await this._paypalCommerceRequestSender.updateOrder({
+                availableShippingOptions: consignment.availableShippingOptions,
+                cartId: cart.id,
+                selectedShippingOption: consignment.selectedShippingOption,
+            });
+        } catch (_error) {
+            throw new RequestError();
+        }
+    }
+
+    private _getAddress(address?: Partial<BillingAddressRequestBody>): BillingAddressRequestBody {
+        return {
+            firstName: address?.firstName || '',
+            lastName: address?.lastName || '',
+            email: address?.email || '',
+            phone: '',
+            company: '',
+            address1: address?.address1 || '',
+            address2: '',
+            city: address?.city || '',
+            countryCode: address?.countryCode || '',
+            postalCode: address?.postalCode || '',
+            stateOrProvince: '',
+            stateOrProvinceCode: address?.stateOrProvinceCode || '',
+            customFields: [],
+        };
+    }
+
+    private _getShippingOptionOrThrow(selectedShippingOptionId?: string): ShippingOption {
+        const state = this._store.getState();
+        const consignment = state.consignments.getConsignmentsOrThrow()[0];
+
+        const availableShippingOptions = consignment.availableShippingOptions || [];
+
+        const recommendedShippingOption = availableShippingOptions.find(
+            (option) => option.isRecommended,
+        );
+        const selectedShippingOption = selectedShippingOptionId
+            ? availableShippingOptions.find((option) => option.id === selectedShippingOptionId)
```

Comment:
Will this return the same `selectedShippingOption`?

Reply:
`selectedShippingOptionId` - is an id of the shipping option provided by PayPal.

In this case it will return an available shipping option data with id provided by PayPal if it exist. In another case, when PayPal didn't provides a `selectedShippingOptionId` we should use previous selected shipping options from consignment object.

Reply:
Thanks for the explanation, didn't realise that `selectedShippingOptionId` was an id returned from paypal.